packages/react-virtual/src/index.tsx

```diff
@@ -50,7 +50,7 @@ function useVirtualizerBase<
     return instance._willUpdate()
   })
 
-  return instance
+  return React.useMemo(() => Object.create(instance), [instance.getVirtualItems()])
```

Comment:
We could move this function inside the virtual core and use the `memo` helper to invalidate the instance generation.

Should give us more flexibility and will be easier to cover more invalidations.